From d62e32a3d4e7d29329a90dd504435253d0359ad9 Mon Sep 17 00:00:00 2001
From: Yanjiang Jin <yanjiang.jin@windriver.com>
Date: Wed, 24 Sep 2014 13:52:07 +0800
Subject: [PATCH 1/2] cav-octeon3: replace cvmx_fpa_free with
 cvmx_fpa3_free_gaura

cvmx_fpa_free has been deprecated in SDK 3.11 525.

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/net/ethernet/octeon/octeon-kexec-net.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/octeon/octeon-kexec-net.c b/drivers/net/ethernet/octeon/octeon-kexec-net.c
index 2ed3507..f347e3c 100644
--- a/drivers/net/ethernet/octeon/octeon-kexec-net.c
+++ b/drivers/net/ethernet/octeon/octeon-kexec-net.c
@@ -29,6 +29,7 @@ void octeon_shutdown_network_hw(void)
 	union cvmx_ipd_ctl_status ipd_ctl_status;
 	union cvmx_pip_sft_rst pip_sft_rst;
 	union cvmx_pow_wq_int wq_int;
+	unsigned aura;
 
 	/* halt the interfaces */
 	num_ints = cvmx_helper_get_number_of_interfaces();
@@ -60,8 +61,9 @@ void octeon_shutdown_network_hw(void)
 
 	/* drain the work queue and free it */
 	while ((work = cvmx_pow_work_request_sync(CVMX_POW_WAIT))) {
+		aura = cvmx_wqe_get_aura(work);
 		cvmx_helper_free_packet_data(work);
-		cvmx_fpa_free(work, CVMX_FPA_WQE_POOL, 0);
+		cvmx_fpa3_free_gaura(work, aura, 0);
 	}
 
 	/* clear possible pending interrupt */
-- 
1.7.5.4

